hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bloom/SparkHoodieBloomIndexHelper.java

```diff
@@ -197,14 +199,15 @@ private Map<HoodieFileGroupId, Long> computeComparisonsPerFileGroup(
       // we will just try exploding the input and then count to determine comparisons
       // FIX(vc): Only do sampling here and extrapolate?
       context.setJobStatus(this.getClass().getSimpleName(), "Compute all comparisons needed between records and files: " + config.getTableName());
-      fileToComparisons = fileComparisonsRDD.countByKey();
+      fileToComparisons = fileComparisonsRDD.countApproxDistinctByKey(0.05).collectAsMap();
```

Comment:
Since we're just using the counts for the partitioning, we can use an approximate and still get the fanout we need

Reply:
I am afraid this might change the dag on spark task retries.
can we move this fix as a new patch. try to do some benchmarking and then we can open it up for review

Reply:
Sounds good to me, I will remove this